plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-order-data.php

```diff
@@ -192,15 +190,19 @@ public static function output( $post ) {
 					$meta_list = array();
 
 					if ( $payment_method && 'other' !== $payment_method ) {
-						/* translators: %s: payment method */
 						$payment_method_string = sprintf(
+							/* translators: %s: payment method */
 							__( 'Payment via %s', 'woocommerce' ),
 							esc_html( isset( $payment_gateways[ $payment_method ] ) ? $payment_gateways[ $payment_method ]->get_title() : $payment_method )
 						);
 
-						if ( $transaction_id = $order->get_transaction_id() ) {
-							if ( isset( $payment_gateways[ $payment_method ] ) && ( $url = $payment_gateways[ $payment_method ]->get_transaction_url( $order ) ) ) {
-								$payment_method_string .= ' (<a href="' . esc_url( $url ) . '" target="_blank">' . esc_html( $transaction_id ) . '</a>)';
+						$transaction_id = $order->get_transaction_id();
+						if ( $transaction_id ) {
+							if ( isset( $payment_gateways[ $payment_method ] ) ) {
+								$url = $payment_gateways[ $payment_method ]->get_transaction_url( $order );
+								if ( $url ) {
+									$payment_method_string .= ' (<a href="' . esc_url( $url ) . '" target="_blank">' . esc_html( $transaction_id ) . '</a>)';
+								}
```

Comment:
If `$url` is empty, `$payment_method_string` should still have the transaction ID, as that's how it was working before:

https://github.com/woocommerce/woocommerce/blob/ecd17484bbd2403fd0d07941e3269dee1f4dfa1b/plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-order-data.php#L202-L206

We could add an `else` branch here or maybe just set `$url` to something like `$url = isset( $payment ... ) ? ...->get_transaction_url() : ''` so that only one `if` is necessary.

Reply:
Ok, did it in a slightly different way because assignment in conditions are generally discouraged.